awx/main/scheduler/dependency_graph.py

```diff
@@ -72,7 +72,12 @@ def project_update_blocked_by(self, job):
         return self.get_item(self.PROJECT_UPDATES, job.project_id)
 
     def inventory_update_blocked_by(self, job):
-        return self.get_item(self.INVENTORY_SOURCE_UPDATES, job.inventory_source_id)
+        inventory_source_block = self.get_item(self.INVENTORY_SOURCE_UPDATES, job.inventory_source_id)
+        if job.inventory_source:
+            project_block = self.get_item(self.PROJECT_UPDATES, job.inventory_source.source_project_id)
+        else:
+            project_block = None
```

Comment:
Still not sure if I agree with this. If the _generated update_ is still running, it will be blocked by

https://github.com/ansible/awx/blob/6b5aa1ec74630b27fd0125a4e28ce5c9f08530bb/awx/main/scheduler/task_manager.py#L109-L112